Test/Unit/Helper/OrderTest.php

```diff
@@ -737,4 +771,34 @@ public function getTransactionState_APMInitialAuthorized()
         $state = $this->currentMock->getTransactionState($this->transactionMock, $this->paymentMock, NULL);
         $this->assertEquals($state, "cc_payment:pending");
     }
+    
+    /**
+     * @test
+     * @covers ::createOrderInvoice
+     */
+    public function createOrderInvoice_amountWithDifferentDecimals() {
+        $totalInvoiced = 0;
+        $amount = 12.64;
+        $grandTotal = 12.6402;
```

Comment:
I guess you can add here a `@dataProvider someVariable` to manage several cases.
as example we can check behaviour like
    * `$amount = 12`;
    * `$grandTotal = 12.00`;
or
    * `$amount = '12'`;
    * `$grandTotal = 12.00`;

and etc